### PR TITLE
770 Owned Entities tests, PATCH, ZeroOrOne/ExactlyOne Owned relationships

### LINKED ISSUE #770: [Nox.Generators] Owned Entities tests, PATCH, ZeroOrOne/ExactlyOne Owned relationships

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs [hunk 1]
```diff
@@ -281,26 +326,32 @@ public async Task PatchToCountryLocalNames_ShouldUpdateCountryLocalName()
         {
```
Comment (anna-naumova25) [inline src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs:326]: @AndreyDegtyarev  see this test for Patch to Owned Entities, it's working now

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs [hunk 2]
```diff
@@ -251,7 +251,7 @@ private static void GeneratePut(Entity entity, CodeBuilder code, NoxSolution sol
     private static void GeneratePatch(Entity entity, string entityName, string pluralName, CodeBuilder code, NoxSolution solution)
     {
         // Method Patch
-        code.AppendLine($"public virtual async Task<ActionResult<{entity.Name}Dto>> Patch({PrimaryKeysFromRoute(entity, solution)}, [FromBody] Delta<{entityName}UpdateDto> {entity.Name.ToLowerFirstChar()})");
+        code.AppendLine($"public virtual async Task<ActionResult<{entity.Name}Dto>> Patch({PrimaryKeysFromRoute(entity, solution)}, [FromBody] Delta<{entityName}Dto> {entity.Name.ToLowerFirstChar()})");
```
Comment (anna-naumova25) [inline src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs:254]: @AndreyDegtyarev see this change that will make deserialization possible

File: src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs [hunk 3]
```diff
@@ -271,6 +280,42 @@ public async Task PutToCountryLocalNames_ShouldUpdateCountryLocalName()
...
+            var getCountryResponse = await _oDataFixture.GetODataSimpleResponseAsync<CountryDto>($"{EntityUrl}/{postCountryResponse!.Id}");
+
+            //Assert
+            putCountryResponse.Should().NotBeNull();
+            putCountryResponse!.Id.Should().Be(postCountryResponse!.Id);
+            putCountryResponse!.Name.Should().Be(postCountryResponse!.Name);
+
+            getCountryResponse!.Id.Should().Be(postCountryResponse!.Id);
+            getCountryResponse!.Name.Should().Be(postCountryResponse!.Name);
+            //getCountryResponse!.CountryBarCode!.BarCodeName.Should().Be(expectedBarCode);
```
Comment (anna-naumova25) [inline src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs:315]: @rochar need to fix PUT and PATCH for single owned entities, it doesn't work now - #772

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs (hunk 2 above)
Comment (AndreyDegtyarev) [reply src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs:254]: Thanks Anna, I will try

**Review by AndreyDegtyarev — APPROVED** [review]

File: src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs (hunk 3 above)
Comment (rochar) [reply src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs:315]: ok

**Review by rochar — APPROVED** [review]

**Pull request merged**